artemis-protocols/artemis-proton-plug/src/main/java/org/proton/plug/util/DeliveryUtil.java

```diff
@@ -30,6 +30,9 @@ public static int readDelivery(Receiver receiver, ByteBuf buffer) {
       while ((count = receiver.recv(buffer.array(), buffer.arrayOffset() + buffer.writerIndex(), buffer.writableBytes())) > 0) {
          // Increment the writer index by the number of bytes written into it while calling recv.
          buffer.writerIndex(buffer.writerIndex() + count);
+         if (buffer.writableBytes() <= 0) {
```

Comment:
shouldn't this be if (buffer.writableBytes() < count)

you may get into a situation that you have space for just part of the buffer coming.